src/hotspot/share/code/codeCache.cpp

```diff
@@ -1416,51 +1389,38 @@ void CodeCache::make_marked_nmethods_deoptimized() {
     CompiledMethod* nm = iter.method();
     if (nm->is_marked_for_deoptimization() && !nm->has_been_deoptimized() && nm->can_be_deoptimized()) {
       nm->make_not_entrant();
-      make_nmethod_deoptimized(nm);
+      nm->make_deoptimized();
```

Comment:
`CodeCache::make_marked_nmethods_deoptimized()` used to call
`make_nmethod_deoptimized(nm)` which made a couple of checks
before it called `nm->make_deoptimized()`. `make_deoptimized()`
does not make these checks so why are they no longer needed?

Here's the removed code:
```

void CodeCache::make_nmethod_deoptimized(CompiledMethod* nm) {
  if (nm->is_marked_for_deoptimization() && nm->can_be_deoptimized()) {
    nm->make_deoptimized();
  }
}
```

Reply:
If the method can't be deoptimized we should not make "not entrant".
If it is already deoptimize no need to try to make it "not entrant" once more.
If it's not marked we shouldn't make it "not entrant".
So only if we make it "not entrant" we should also make_deopt().

This is checked at the line just before:
```
if (nm->is_marked_for_deoptimization() && !nm->has_been_deoptimized() && nm->can_be_deoptimized()) {
      nm->make_not_entrant();
      nm->make_deoptimized();
}
```

So we do not need to recheck it.

But I'll add a check that we do not re-patch the nops.

Reply:
Thanks!